### PR TITLE
fix: restrict backdated leave applications for specified role

### PR DESCRIPTION
**Pull request by vishdha**
Changes made:

Check for backdated leave application added
Only HR Manager can make backdated leave applications

### CODE REVIEW

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/hr/doctype/hr_settings/hr_settings.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=4768405&bid=15893431)

File: erpnext/hr/doctype/hr_settings/hr_settings.js [hunk 1]
```diff
@@ -19,5 +19,8 @@ frappe.ui.form.on('HR Settings', {
 			}
 			frm.set_value('password_policy', policy.split(new RegExp(" |-", 'g')).filter((token) => token).join('-'));
 		}
+		if(frm.doc.restrict_backdated_leave_application ==1) {
+			frm.set_df_property("role_allowed_to_create_backdated_leave_application", "reqd", 1);
```
Comment (Alchez) [inline erpnext/hr/doctype/hr_settings/hr_settings.js]:
```suggestion
			frm.toggle_reqd(fieldname, frm.doc.restrict_backdated_leave_application)
```

File: erpnext/hr/doctype/leave_application/leave_application.py [hunk 2]
```diff
@@ -79,9 +79,11 @@ def validate_applicable_after(self):
...
 
 	def validate_dates(self):
-		if self.from_date and self.from_date < frappe.utils.today():
-			if "HR Manager" not in frappe.get_roles():
-				frappe.throw(_("Only HR Managers can create backdated leave applications"))
+		if frappe.db.get_single_value("HR Settings", "restrict_backdated_leave_application"):
+			if self.from_date and self.from_date < frappe.utils.today():
+				allowed_role = frappe.db.get_single_value("HR Settings", "role_allowed_to_create_backdated_leave_application")
+				if allowed_role not in frappe.get_roles():
+					frappe.throw(_("Only {0} can create backdated leave applications").format(allowed_role))
```
Comment (Alchez) [inline erpnext/hr/doctype/leave_application/leave_application.py]:
```suggestion
					frappe.throw(_("Only users with the {0} role can create backdated leave applications").format(allowed_role))
```

**Comment by vishdha** [comment]
https://github.com/frappe/erpnext/pull/20201

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DigiThinkIT/erpnext/pullRequest?prid=4768405&bid=15893431)

**Review by Alchez — APPROVED** [review]

**Pull request merged**